### DISCUSSION TITLE
Replace Colord library

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco.UI#938] [General]
When starting building the color picker I chosen Colord library with modular system, but since then it hasn't really been maintained.

I wonder if we should consider replacing with a different library instead.

Some features it should have:
- Lightweight
- Modular
- Convert colors between HEX, RGB, HSL, HSV, HWB
- Convert color name, e.g. red, green, blue
- TypeScript support

Suggestions:
- [Colorjs.io](https://colorjs.io/)
- [chroma.js](https://gka.github.io/chroma.js/)
- [Culorijs](https://culorijs.org/)
- [color-bits](https://romgrk.com/posts/color-bits/)

**Comment by bjarnef** [comment]
@iOvergaard @nielslyngsoe should we consider a different library? The existing [Colord](https://www.npmjs.com/package/colord) library haven't had any releases for 2 years and seems to be unmaintained.

I also noticed a library called [color-bits](https://www.npmjs.com/package/color-bits) even faster than the existing Colord library.
It has some [functions](https://github.com/romgrk/color-bits/blob/master/docs/README.md#functions) but I am unsure if it support all of the current feature like converting color name to e.g. hex color.

**Comment by bjarnef** [comment]
@iOvergaard @nielslyngsoe do you have any thought on this? The Colord library seems to be unmaintained and no new releases for 2½ years, so should we consider one of the alternatives mentioned?

**Reply by iOvergaard** [reply]
We want to investigate this for version 2.0. Would you happen to have an idea as to how we can replace it?

**Reply by bjarnef** [reply]
Most other color libraries have similar functions for color converting. So I think it should be simple enough just to swap, but consider the different features.
https://github.com/umbraco/Umbraco.UI/discussions/938#discussion-7339210

**Reply by iOvergaard** [reply]
Swapped with culori in the upcoming UUI 2.0, which seems well-maintained.

**Comment by iOvergaard** [comment]
Swapped with culori in the upcoming UUI 2.0, which seems well-maintained.
